lib/longleaf/helpers/s3_uri_helper.rb

```diff
@@ -31,6 +31,30 @@ def self.extract_bucket(url)
         return prefix[0..-2]
       end
     end
+    
+    def self.extract_path(url)
+      uri = s3_uri(url)
+
+      matches = ENDPOINT_PATTERN.match(uri.host)
+      if matches.nil?
+        raise ArgumentError.new("Provided URI does match the expected pattern for an S3 URI")
+      end
+
+      path = uri.path
+      return nil if path == '/'
```

Comment:
Could there be an empty path?

Reply:
I don't think this should be able to occur in the current usage, since location paths are forced to end in slashes. But in general if the path didn't end with a slash, then the path would be empty, so I will add that case in.